src/oscar/apps/dashboard/catalogue/forms.py

```diff
@@ -27,11 +27,27 @@
 
 BaseCategoryForm = movenodeform_factory(
     Category,
-    fields=['name', 'slug', 'description', 'image', 'is_public'],
+    fields=['name', 'slug', 'description', 'image', 'is_public', 'meta_title', 'meta_description'],
     exclude=['ancestors_are_public'])
 
 
-class CategoryForm(BaseCategoryForm):
+class SEOFormMixin:
+    seo_fields = ['meta_title', 'meta_description', 'slug']
+
+    def primary_form_fields(self):
+        return [field for field in self if not field.is_hidden and not self.is_seo_field(field)]
+
+    def seo_form_fields(self):
+        return [field for field in self if self.is_seo_field(field)]
+
+    def is_seo_field(self, field):
+        return field.name in self.seo_fields
+
+
+class CategoryForm(SEOFormMixin, BaseCategoryForm):
+    meta_description = forms.CharField(required=False, label=_('Meta description'), widget=forms.Textarea(attrs={
```

Comment:
This could be moved into `SEOFormMixin` to avoid duplication.

Reply:
With the move, I would lose the `no-widget-init`  effect on the widget.

Reply:
OK - looks like the modelform is cobbering what it defined in the mixin. Having looked a bit closer I think we can define the widget on the `Meta.widgets` property (see line 225 below), and then we don't have to redefine the field itself at all:

```
'meta_description': forms.Textarea(attrs={'class': 'no-widget-init'})
```

You can do the same on the category form.